sous_chef/sous_chef/menu/create_menu.py

```diff
@@ -158,39 +166,80 @@ def _inspect_unrated_recipe(self, recipe: Recipe):
                 print(recipe.ingredient_field)
 
     def _load_fixed_menu(self, gsheets_helper):
+        menu_basic_file = self.config.fixed.basic
+        menu_basic = gsheets_helper.get_sheet_as_df(
+            menu_basic_file, menu_basic_file
+        )
+
         menu_number = self._check_fixed_menu_number(
             self.config.fixed.menu_number
         )
-        menu_file = f"{self.config.fixed.file_prefix}{menu_number}"
-        fixed_menu = gsheets_helper.get_sheet_as_df(menu_file, menu_file)
-        return fixed_menu
+        menu_fixed_file = f"{self.config.fixed.file_prefix}{menu_number}"
+        menu_fixed = gsheets_helper.get_sheet_as_df(
+            menu_fixed_file, menu_fixed_file
+        )
+        return pd.concat([menu_basic, menu_fixed]).sort_values(
+            by=["weekday", "meal_time"]
+        )
 
     def _load_local_menu(self):
         file_path = Path(ABS_FILE_PATH, self.config.local.file_path)
         # fillna("") to keep consistent with gsheets implementation
         return pd.read_csv(file_path, sep=";").fillna("")
 
-    def _retrieve_manual_menu_ingredient(self, row: pd.Series):
-        ingredient = self.ingredient_formatter.format_manual_ingredient(
-            quantity=float(row["eat_factor"]),
-            unit=row["eat_unit"],
+    def _process_menu(self, row: pd.Series):
+        # due to schema validation row["type"], may only be 1 of these 4 types
+        FILE_LOGGER.info(
+            "[process menu]",
+            action="processing",
+            day=row["weekday"],
             item=row["item"],
+            type=row["type"],
         )
+        if row["type"] == "category":
+            return self._select_random_recipe(
+                row, "get_random_recipe_by_category"
+            )
+        if row["type"] == "ingredient":
+            # do NOT need returned as found as is
+            self._check_manual_ingredient(row)
+            return row
+        if row["type"] == "recipe":
+            return self._check_recipe_and_add_cooking_time(row)
+        if row["type"] == "tag":
+            return self._select_random_recipe(row, "get_random_recipe_by_tag")
+
+    def _retrieve_manual_menu_ingredient(
+        self, row: pd.Series
+    ) -> MenuIngredient:
         return MenuIngredient(
```

Comment:
Would it be not too much effort to switch MenuRecipe, Recipe, MenuIngredient, and Ingredient to a schema at this stage? We could only do one of the pairs to get that started. What do you think?

Reply:
Would do another time. Ticket has already grown quite a bit since this was written.